src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -161,14 +162,24 @@ function _is_action_fit(method, statement) {
     return statement.Action ? action_fit : !action_fit;
 }
 
-function _is_principal_fit(account, statement) {
+function _is_principal_fit(identifier_object, statement) {
+    // We changed the account (which was account name as string) to be identifier_object (object)
+    // In case someone calls it with account name, we will wrap it in an object
+    if (typeof identifier_object !== 'object') identifier_object = { name: identifier_object };
     let statement_principal = statement.Principal || statement.NotPrincipal;
 
     let principal_fit = false;
     statement_principal = statement_principal.AWS ? statement_principal.AWS : statement_principal;
     for (const principal of _.flatten([statement_principal])) {
-        dbg.log1('bucket_policy: ', statement.Principal ? 'Principal' : 'NotPrincipal', ' fit?', principal, account);
-        if ((principal.unwrap() === '*') || (principal.unwrap() === account)) {
+        dbg.log1('bucket_policy: ', statement.Principal ? 'Principal' : 'NotPrincipal', ' fit?', principal, identifier_object);
+        const principal_str = principal.unwrap();
+        let account_name;
+        if (identifier_object?.name) {
+            account_name = identifier_object?.name instanceof SensitiveString ? identifier_object?.name.unwrap() : identifier_object?.name;
+        }
+        if ((principal_str === '*') ||
+            (principal_str === account_name) ||
```

Comment:
as we discussed, let's go with Amit's suggestion - https://github.com/noobaa/noobaa-core/pull/8167/files#diff-11cc61806e3f6607fe678e081751019f21db0f09d48a56a26713203b3ada1db5R263-R274